GLTFSDK/Source/ExtensionsKHR.cpp

```diff
@@ -367,3 +371,110 @@ std::unique_ptr<Extension> KHR::MeshPrimitives::DeserializeDracoMeshCompression(
 
     return extension;
 }
+
+// KHR::TextureInfos::TextureTransform
+
+KHR::TextureInfos::TextureTransform::TextureTransform() :
+    offset(0.0f, 0.0f),
+    rotation(0.0f),
+    scale(1.0f, 1.0f),
+    texCoord(0)
+{
+}
+
+std::unique_ptr<Extension> KHR::TextureInfos::TextureTransform::Clone() const
+{
+    return std::make_unique<TextureTransform>(*this);
+}
+
+bool KHR::TextureInfos::TextureTransform::IsEqual(const Extension& rhs) const
+{
+    const auto other = dynamic_cast<const TextureTransform*>(&rhs);
+
+    return other != nullptr
+        && glTFProperty::Equals(*this, *other)
+        && this->offset == other->offset
+        && this->rotation == other->rotation
+        && this->scale == other->scale
+        && this->texCoord == other->texCoord;
+}
+
+std::string KHR::TextureInfos::SerializeTextureTransform(const TextureTransform& textureTransform, const Document& /*gltfDocument*/)
+{
+    rapidjson::Document doc;
+    auto& a = doc.GetAllocator();
+    rapidjson::Value KHR_textureTransform(rapidjson::kObjectType);
+    {
+        if (textureTransform.offset != Vector2(0.0f, 0.0f))
+        {
+            KHR_textureTransform.AddMember("offset", RapidJsonUtils::ToJsonArray(textureTransform.offset, a), a);
+        }
+
+        if (textureTransform.rotation != 0.0f)
+        {
+            KHR_textureTransform.AddMember("rotation", textureTransform.rotation, a);
+        }
+
+        if (textureTransform.scale != Vector2(1.0f, 1.0f))
+        {
+            KHR_textureTransform.AddMember("scale", RapidJsonUtils::ToJsonArray(textureTransform.scale, a), a);
+        }
+
+        if (textureTransform.texCoord != 0)
+        {
+            KHR_textureTransform.AddMember("texCoord", ToKnownSizeType(textureTransform.texCoord), a);
+        }
+
+        SerializeProperty(textureTransform, KHR_textureTransform, a);
+    }
+
+    glTF::rapidjson::StringBuffer buffer;
+    glTF::rapidjson::Writer<glTF::rapidjson::StringBuffer> writer(buffer);
+    KHR_textureTransform.Accept(writer);
+
+    return buffer.GetString();
+}
+
+std::unique_ptr<Extension> KHR::TextureInfos::DeserializeTextureTransform(const std::string& json)
+{
+    TextureTransform textureTransform;
+
+    auto doc = RapidJsonUtils::CreateDocumentFromString(json);
+    const rapidjson::Value sit = doc.GetObject();
+
+    // Offset
+    auto offsetIt = sit.FindMember("offset");
+    if (offsetIt != sit.MemberEnd())
+    {
+        std::vector<float> offset;
+        for (rapidjson::Value::ConstValueIterator ait = offsetIt->value.Begin(); ait != offsetIt->value.End(); ++ait)
+        {
+            offset.push_back(static_cast<float>(ait->GetDouble()));
+        }
+        // NOTE: Schema validation will handle this once the extension is added to the GLTF SDK
+        textureTransform.offset = (offset.size() == 2) ? Vector2(offset[0], offset[1]) : Vector2(0.0f, 0.0f);
```

Comment:
Wouldn't throwing an exception be the better option if the array isn't exactly two elements in size?

Reply:
Also that comment "Schema validation will handle this once the extension is added to the GLTF SDK" no longer makes sense since this PR is into the SDK.